### PR TITLE
Fix "non responsive buttons" issue on the XO

### LINKED ISSUE #42: Resolve "non responsive buttons" issue on the XO
Nothing happens when you click buttons. It works properly on a regular machine, however when running on an XO nothings happens.

### LINKED ISSUE #43: New version of Sugargame available
Hey, I see you're using Sugargame 1.1, there's a new 1.2 [available](https://github.com/sugarlabs/sugargame) which has a few fixes.

### PR DESCRIPTION
**Pull request by quozl**
Fixes #42.

Fixes #43.

Gtk+ events were not being handled in the application event loop, so we
add a callback to check for and handle the events.

Also ported to Sugargame v1.2 which solves some other problems.

Tested on OLPC XO-4 with OLPC OS 13.2.9.

### CODE REVIEW

**Review by justwheel — APPROVED** [review]
Wow, so first off, thanks so much for the contribution, @quozl! 😄 

As a caveat, my approval is not a deep technical analysis. I'll follow up with @wilfriedE to see if he can take a quick peek here just so he's up-to-date on what's happening.

I also noticed you said you tested on an XO-4. The XOs at RIT (that we use for development) are mostly XO-1's. I'm in favor of bringing the game up-to-date to be useful for the current generation, but I also wonder if there will be compatibility issues – it may warrant a new major release. @quozl: What do you think?

**Comment by ghost** [comment]
On Fri, Jan 19, 2018 at 11:42:35AM +0000, Justin W. Flory wrote:

> I also noticed you said you tested on an XO-4. The XOs at RIT [...]

In the semester in which PyCut was originally developed, we did only target
RIT's XOs.  Starting a year ago, using RIT's XOs was the default, but we
added the option for teams to design for Sugarizer.  Sugarizer was embraced
with some success by two out of three teams, each of which had strong
advocates for targetting that kind of environment.

This semester, we have local support for using Linux-based virtual machines
atop the default Windows desktop in the provided class space.[1] In this
light, our default development environment and target will go back to being
Python-based Sugar.  For some non-Sugar exercises at the very least we'll
use what we can from the latest Sugar-on-a-Stick installed on a networked
hosting platform.

To the extent SoaS presents obstacles, we can install other things in VMs on
the classroom machines, not the least of which includes my plan to introduce
Sugar Live Build:

  * https://wiki.sugarlabs.org/go/0.112#How_to_test.3F_2

Students also will have the option of installing atop VMs on whatever
personal machines they can.  My hope is that our demonstrating in class how
to do such installations will go a long way towards empowering students in
that direction, setting them up for a sustainable mode of individual
participation in this and other free and open source projects.

We will retain Sugarizer as an option, but since we use Python elsewhere and
the community resources for Sugar are more cleanly modular (activity code
tracked and distributed separately from other activities and from the
"runtime") I'm gladly turning our attention more fully back to Sugar.

For now, students who remain interested in our XO's unique affordances will
still have the opportunity to target Sugar as it runs on them, but with my
making as clearly as I can what significant and limiting constraints are
currently involved.[2] 

Bottom line: If we have much success at all this semester, I think we can
afford to let PyCut move forward with Sugar in one form or another.

Thanks to the PyCut team and to James Cameron for your ongoing attention and
efforts!


[1] I did start down the path of seeing what it would take to run Sugar with
Windows-native dependencies but concluded this was not something I had the
ability or support to make work.

[2] Perhaps we can incorporate the hardware in future in some other
exercise, because its design remains unsurpassed in many ways.

**Comment by quozl** [comment]
Thanks for the comments.

I've just now tested on XO-1, and behaviour is identical except for slightly lower responsiveness to button clicks.  That's expected.

I had also tested on Ubuntu 18.04 beta with Sugar 0.112.

Have not tested on Sugar Live Build, but expect it to work, as the `python-pygame` package is included.

If you need further explanation for the changes in the patch, I can break the patch into multiple commits or annotate the commits here.  Your call.

**Review by wilfriedE — APPROVED** [review]
Tested it out and works well!! @quozl Thanks again for making this fix. 
@jflory7 I am going to merge this and I think we can go ahead and create a new release!
